### PR TITLE
Automate HIP Management with Number Assignment and Status Updates

### PR DESCRIPTION
**Pull request by mgarbs**
### Description:
This PR introduces two automated GitHub Actions workflows to streamline the management of HIPs (Hedera Improvement Proposals):

1. **Assign HIP Number and Rename File:**
    - This workflow identifies new HIP proposals by comparing the files modified in a pull request (PR) against the `main` branch.
    - If a new HIP is detected (a proposal that doesn't yet exist in the `main` branch), the workflow assigns a unique HIP number based on the PR number.
    - It renames the file following the standard format (`hip-<PR_NUMBER>.md`) and updates the `hip:` field in the document header with the assigned number.
    - Finally, it commits the changes back to the PR branch for a clear, consistent HIP file structure.

2. **Schedule Status Update Based on Last Call Time:**
    - This workflow automatically updates the status of HIPs that have reached their last call deadline.
    - It identifies merged PRs containing HIP files marked with a `status: Last Call` field.
    - The workflow calculates the appropriate delay until the specified `last-call-date-time`, and then applies the correct new status based on the HIP type and category.
    - It also adds or updates the `updated` field with the current date.
    - After making these adjustments, it creates a new PR for the changes and automatically merges it.

### Benefits:
- **Automation:** Streamlines HIP management, saving time and reducing manual errors.
- **Consistency:** Ensures all new and updated HIPs adhere to a consistent naming and status convention.
- **Transparency:** Offers clear communication through automatic PR messages and the precise updating of statuses and dates.

### CODE REVIEW

**Review by sergmetelin — APPROVED** [review]
LGTM

File: .github/workflows/add-hip-number.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
+name: Assign HIP Number and Rename File
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  assign-hip-number:
+    runs-on: ubuntu-latest
```
Comment (rbarkerSL) [inline .github/workflows/add-hip-number.yml:9]: should be on the self hosted runner

`runs-on: [self-hosted, Linux, medium, ephemeral`

File: .github/workflows/add-hip-number.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    steps:
+      - name: Check out the code
```
Comment (rbarkerSL) [inline .github/workflows/add-hip-number.yml:11]: Need to add a hardened runner step:

```
      - name: Harden Runner
        uses: step-security/harden-runner@a4aa98b93cab29d9b1101a6143fb8bce00e2eac4 # v2.7.1
        with:
          egress-policy: audit
 ```

File: .github/workflows/add-hip-number.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        uses: actions/checkout@v2
```
Comment (rbarkerSL) [inline .github/workflows/add-hip-number.yml:12]: `uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0`

File: .github/workflows/add-hip-number.yml [hunk 4]
```diff
@@ -0,0 +1,97 @@
+name: Assign HIP Number and Rename File
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
```
Comment (rbarkerSL) [inline .github/workflows/add-hip-number.yml:6]: need to add defaults and permissions:

```
defaults:
    run:
        shell: bash
```
 
 ```
 permissions:
     contents: write
     issues: read
     pull-requests: write
     checks: write
```

File: .github/workflows/schedule-last-call-date-end.yml [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+permissions:
+  pull-requests: write
+  contents: write
+
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+
```
Comment (rbarkerSL) [inline .github/workflows/schedule-last-call-date-end.yml:12]: need defaults:

```
defaults:
    runs:
        shell
```

File: .github/workflows/schedule-last-call-date-end.yml [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+jobs:
+  check-merged-file:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
```
Comment (rbarkerSL) [inline .github/workflows/schedule-last-call-date-end.yml:16]: Use the self-hosted runner:

`runs-on: [self-hosted, Linux, medium, ephemeral]`

File: .github/workflows/schedule-last-call-date-end.yml [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+    steps:
+      - name: Check out the code
+        uses: actions/checkout@v2
```
Comment (rbarkerSL) [inline .github/workflows/schedule-last-call-date-end.yml:20]: Actions should be using commit strings as versions:

`uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0`

File: .github/workflows/schedule-last-call-date-end.yml [hunk 8]
```diff
@@ -0,0 +1,162 @@
+name: Schedule Status Update Based on Last Call Time
+
+permissions:
+  pull-requests: write
+  contents: write
```
Comment (rbarkerSL) [inline .github/workflows/schedule-last-call-date-end.yml:5]: Add `packages: write` permisison

**Review by rbarkerSL — CHANGES_REQUESTED** [review]
Couple of comments

File: .github/workflows/add-hip-number.yml (hunk 4 above)
Comment (mgarbs) [reply .github/workflows/add-hip-number.yml:6]: Added

File: .github/workflows/add-hip-number.yml (hunk 2 above)
Comment (mgarbs) [reply .github/workflows/add-hip-number.yml:26]: Added

File: .github/workflows/schedule-last-call-date-end.yml (hunk 8 above)
Comment (mgarbs) [reply .github/workflows/schedule-last-call-date-end.yml:5]: Added

File: .github/workflows/schedule-last-call-date-end.yml (hunk 5 above)
Comment (mgarbs) [reply .github/workflows/schedule-last-call-date-end.yml:17]: Added

**Review by rbarkerSL — APPROVED** [review]

**Pull request merged**